src/wms/src/main/java/org/geoserver/wms/map/RenderedImageMapOutputFormat.java

```diff
@@ -1139,7 +1148,10 @@ protected GridCoverage2D readCoverage(GridCoverage2DReader reader,
                     }
                 } else {
                     if (transparent) {
-                        image = addAlphaChannel(image);
+                        final ImageWorker iw = new ImageWorker(image);
```

Comment:
Wouldn't it be better to have back a addAlphaChannel method, and avoid having two image workers with different names in the same context? (worker and iw)